test/unit_test/model/test_onnx_model.py

```diff
@@ -39,6 +39,161 @@ def test_model_prepare_session(mock_get_available_providers, inference_session_m
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [
+                    {"device_id": "0", "tunable_op_enable": "True", "tunable_op_tuning_enable": "True"}
+                ],
+            },
+        ),
+        # parameter inference_settings cases
+        (
+            {
+                "execution_provider": [("ROCMExecutionProvider", {"device_id": 0})],
+            },
+            None,
+            None,
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": "0"}],
+            },
+        ),
+        (
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": 1}],
+            },
+            None,
+            None,
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": "1"}],
+            },
+        ),
+        # model inference_settings cases
+        (
+            None,
+            {
+                "execution_provider": [("ROCMExecutionProvider", {"device_id": 2})],
+            },
+            None,
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": "2"}],
+            },
+        ),
+        (
+            None,
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": 3}],
+            },
+            None,
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": "3"}],
+            },
+        ),
+        # parameter inference_settings and model inference_settings cases
+        (
+            {
+                "execution_provider": [("ROCMExecutionProvider", {"device_id": 4})],
+            },
+            {
+                "execution_provider": ["MIGraphXExecutionProvider"],
+            },
+            None,
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": "4"}],
+            },
+        ),
+        (
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": 5}],
+            },
+            {
+                "execution_provider": [("MIGraphXExecutionProvider", {})],
+            },
+            None,
+            {
+                "execution_provider": ["ROCMExecutionProvider"],
+                "provider_options": [{"device_id": "5"}],
+            },
+        ),
+    ],
+)
+@patch("onnxruntime.InferenceSession")
+@patch("onnxruntime.get_available_providers")
+def test_model_prepare_session_multiple_inference_settings(
+    mock_get_available_providers,
+    inference_session_mock,
```

Comment:
Better to align the name mock_* or *_mock.

**mock_**get_available_providers,
inference_session**_mock**,